mdakit_membcurv/tests/test_mdakit_membcurv.py

```diff
@@ -3,10 +3,290 @@
+                                                1.30710748e-03, -3.49076390e-03, -1.28226749e-02,
+                                                -1.41168166e-02, -1.08870941e-02, -1.21323346e+00,
+                                                -1.55778010e-01]])}
+}
+
+
+@pytest.fixture()
+def universe():
+    u = mda.Universe(GRO_MEMBRANE_PROTEIN, XTC_MEMBRANE_PROTEIN)
+    return u
+
+
+@pytest.fixture()
+def ref_beads():
+    u = mda.Universe(GRO_PO4)
+    ref_beads = u.atoms[0:455]
+    return ref_beads
+
+
+@pytest.fixture()
+def Z_cloud(ref_beads):
+    n_cells, max_width = 10, 19
+
+    # set grid_1. z coordinates for [i,j] cells.
+    # set grid_2. Count number of beads populating each [i,j] cell.
+    # set z_ref. Final values of average z per [i,j] cell.
+    grid_1, grid_2, z_ref = [np.zeros([n_cells, n_cells]) for i in range(3)]
+
+    # factor used to map (x,y) to [i,j]
+    factor = np.float32(n_cells / max_width)
+
+    # iterate over the bead of reference
+    for atom in ref_beads:
+
+        # extract positions in nm.
+        x, y, z = atom.position / 10
+
+        # map (x,y) to [i,j]
+        cell_l = int(abs(x) * factor)
+        cell_m = int(abs(y) * factor)
+
+        # sum value of z coordinate
+        grid_1[cell_l, cell_m] += z
+        grid_2[cell_l, cell_m] += 1
+
+    # To calculate average, iterate over each cell in the grid:
+    for i, j in it.product(range(n_cells), range(n_cells)):
+        # if the element [i,j] is not empty
+        if grid_2[i, j] > 0:
+            # then calculate the average of z:
+            # grid_1 has the sum of all the z coordinates.
+            # grid_2 counted how many beads were in that grid.
+            z_ref[i, j] += grid_1[i, j] / grid_2[i, j]
+        else:
+            # if there are not beads in that cell, store a nan.
+            z_ref[i, j] = np.nan
+
+    return z_ref
+
+
+@pytest.fixture()
+def mdtraj_po4():
+    # trajectory PO4 beads only imported using mdtraj, gone after refactoring
+    mdtraj = md.load(XTC_PO4, top=GRO_PO4)
+    return mdtraj
+
+
+@pytest.fixture(scope="session")
+def output(tmpdir_factory):
+    tmp = tmpdir_factory.mktemp("output")
+    return tmp
+
+
+@pytest.fixture()
+def md_ref_beads():
+    # reference beads using mdtraj, gone after refactoring. Select upper leaflet
+    topology = md.load(GRO_PO4).topology
+    md_ref_beads = {'upper': {'POPC': topology.select('resname POPC and index 1 to 457').astype(int).tolist()}}
+    return md_ref_beads
+
 
 def test_mdakit_membcurv_imported():
     """Sample test, will always pass so long as import statement worked"""
     assert "mdakit_membcurv" in sys.modules
+
+
+def test_md_topology():  # line 92, gone after refactoring
+    top = md.load(GRO_MEMBRANE_PROTEIN).topology
+    assert top.n_atoms == 18891
+
+
+def test_box_size(universe):  # line 99
+    box_size = universe.dimensions[0]
+    assert_almost_equal(box_size, 184.31013, decimal=5)
+
+
+def test_n_cells(universe):  # line 101
```

Comment:
`test_box_size, test_dict_to_pickle, test_md_topology` don't test *your* code so they should get removed